### PR TITLE
Add course-staff field in the registrar read course enrollment api response

### PR DESCRIPTION
**Pull request by how015**
MST-188: Add course-staff field in the response of read course enrollment api in registrar.

As the first step, add a unit test for feedback

### CODE REVIEW

**Review by kdmccormick — CHANGES_REQUESTED** [review]
From the ticket:

>    The GET course enrollments endpoint should return an additional body field course_staff
>    The value should match that returned from LMS

These tests catch that :+1: 

> If role management waffle not enabled for the requesting user the course_staff field should not be returned

How about this, though?

File: registrar/apps/api/v1/tests/test_views.py [hunk 1]
```diff
@@ -1344,6 +1367,42 @@ def test_ok(self, format_param, expected_format, expected_contents, _enrollments
...
 
+    @patch_discovery_program_details(mock_program_details)
+    @mock.patch.object(lms, 'get_course_run_enrollments', return_value=enrollments_with_course_staff)
+    @ddt.data(
+        (None, 'json', enrollments_with_course_staff_json),
+        ('json', 'json', enrollments_with_course_staff_json),
+        ('csv', 'csv', enrollments_with_course_staff_csv),
+    )
+    @ddt.unpack
+    def test_ok_with_course_staff(self, format_param, expected_format, expected_contents, _enrollments_mock):
```
Comment (schenedx) [inline registrar/apps/api/v1/tests/test_views.py]: So, the only thing about this test is that, it does not test the waffle flag on or off.
With waffle flag off, even if the LMS API returns a json with course_staff key, the registrar should not return that on the CSV or JSON file.
Only with waffle flag on, the course_staff key and value should show on the registrar output file. Please incorporate your test with waffle flag.

**Review by schenedx — CHANGES_REQUESTED** [review]
The test setup looks good. I think you just need to incorporate the waffle flag.

File: registrar/apps/api/v1/tests/test_views.py (hunk 1 above)
Comment (how015) [reply registrar/apps/api/v1/tests/test_views.py]: Got it, thank you, Simon. Actually working on it already :)

File: registrar/apps/api/v1/tests/test_views.py [hunk 2]
```diff
@@ -1344,6 +1367,82 @@ def test_ok(self, format_param, expected_format, expected_contents, _enrollments
...
+        format_suffix = "?fmt=" + format_param if format_param else ""
+        kwargs = {'result_format': format_param} if format_param else {}
+        with self.assert_tracking(
+                user=self.hum_admin,
+                program_key='masters-in-english',
+                course_id='HUMx+English-550+Spring',
+                status_code=202,
+                **kwargs
+        ):
+            with activate_waffle_flag('enable_course_role_management', self.stem_admin_group):
```
Comment (schenedx) [inline registrar/apps/api/v1/tests/test_views.py:1427]: If it's waffle disabled, why are you still enabling waffle for a different permission group? That should be a different test case.

File: registrar/apps/enrollments/tasks.py [hunk 3]
```diff
@@ -95,10 +96,12 @@ def list_course_run_enrollments(
         program_key,
         internal_course_key,
         external_course_key,
+        course_role_management_enabled,
```
Comment (schenedx) [inline registrar/apps/enrollments/tasks.py]: Can you please have a default value for this function? It's for deployment backwards compatibility.

**Review by schenedx — CHANGES_REQUESTED** [review]
See inline comments.

File: registrar/apps/api/v1/tests/test_views.py [hunk 4]
```diff
@@ -1344,6 +1367,119 @@ def test_ok(self, format_param, expected_format, expected_contents, _enrollments
...
+
+    @patch_discovery_program_details(mock_program_details)
+    @mock.patch.object(lms, 'get_course_run_enrollments', return_value=enrollments)
+    @ddt.data(
+        (None, 'json', enrollments_json),
+        ('json', 'json', enrollments_json),
+        ('csv', 'csv', enrollments_csv),
+    )
+    @ddt.unpack
+    def test_ok_course_staff_waffle_enabled_on_other_group(self, format_param, expected_format, expected_contents, _enrollments_mock):
```
Comment (schenedx) [inline registrar/apps/api/v1/tests/test_views.py]: #NIT change the title of this function to `test_no_course_staff_waffle_enabled_on_unrelated_group`

**Review by schenedx — APPROVED** [review]
Please fix your quality errors

**Review by schenedx — COMMENTED** [review]
Remember to squash your commits before you merge.

**Comment by how015** [comment]
MST-188

**Pull request merged**